src/test/java/uk/gov/justice/digital/config/BaseSparkTest.java

```diff
@@ -0,0 +1,193 @@
+package uk.gov.justice.digital.config;
+
+
+
+import static org.apache.spark.sql.functions.col;
+import static org.apache.spark.sql.functions.from_json;
+import static org.apache.spark.sql.functions.lit;
+import static org.apache.spark.sql.functions.schema_of_json;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.types.DataType;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.io.TempDir;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.profile.ProfileCredentialsProvider;
+
+public abstract class BaseSparkTest {
+
+	@TempDir
+	Path folder;
+	
+	protected SparkSession spark;
+	
+
+	protected String accessKey;
+	protected String secretKey;
+	
+
+	@BeforeEach
+	public void before() {
+		
+		spark = SparkSession.builder()
+			    .appName("test")
+			    .enableHiveSupport()
+				.config("spark.master", "local")
```

Comment:
Also disable spark ui to speed things up, since it starts up an additional server to serve the UI which we don't need in this context.